### ISSUE TITLE
Use the sha256sum (or whatever) of libsodium.tar.gz instead of the signature

### ISSUE THREAD

**Issue by tasn** [lyubo/react-native-sodium#32]
Thanks for your great work.

I was thinking: why not use the sha256sum (or sha512sum, or whatever) instead of gpg? It removes the need for having gpg installed, the key imported and the tar.gz being different (but still signed with a valid key).

Thanks.

**Comment by lyubo** [comment]
Hi, Tom
This is out of my control. libsodium sources are downloaded from here: https://download.libsodium.org/libsodium/releases/
This is the official download location and it's maintainer decision how to sign library source. They use either gpg or a tool called minisign. These are the two options we have. So, we can switch to minisign if it's easier to install.

**Comment by tasn** [comment]
Yeah, I know, but you can still calculate the sha sum yourself, and just make sure it's what you expect it to be.
For example, here's how they build libsodium on Arch Linux: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/libsodium

This line downloads the sources:
```
source=("https://download.libsodium.org/libsodium/releases/libsodium-$pkgver.tar.gz"{,.minisig})
```

and this one verifies the sha512sum:
```
sha512sums=('17e8638e46d8f6f7d024fe5559eccf2b8baf23e143fadd472a7d29d228b186d86686a5e6920385fe2020729119a5f12f989c3a782afbd05a8db4819bb18666ef'
```

They also verify the minisign signature in Arch, but there's really no need. The sha512sum assures that the file hasn't been tampered with, which is what we want.

So in summary, every time you upgrade libsodium, you also need to calculate the sha512sum and update the verifier script (or wherever this package verifies the signature) to verify the shasum is what you would expect. You can use the `sha512sum` utility (standard on most systems) to achieve it.

I'm happy to further elaborate if you'd like me to. I believe it's the way to go as it's much simpler, slightly more secure, and is how things are done almost everywhere.

**Comment by lyubo** [comment]
I wanted to avoid another piece of work when upgrading libsodium version, but indeed using shasum is easier. So, I would be glad to have your contribution on this one.
